File: src/main/java/com/catan/catanbackend/model/dto/move_dtos/UpgradeStructureDto.java
package com.catan.catanbackend.model.dto.move_dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpgradeStructureDto {
    private Integer tileX;
    private Integer tileY;
    private Integer cornerIndex;
}

File: Assets/Scripts/Dtos/GameMoves/UpgradeStructureDto.cs
using System;

namespace Assets.Scripts.Dtos
{
    [Serializable]
    public class UpgradeStructureDto
    {
        public int tileX;
        public int tileY;
        public int cornerIndex;

        public UpgradeStructureDto(int tileX, int tileY, int cornerIndex)
        {
            this.tileX = tileX;
            this.tileY = tileY;
            this.cornerIndex = cornerIndex;
        }
    }
}
